Add DEMO Config Variable

lhirschfeld opened the pull request:
Introduced 'DEMO' config variable, which decides whether all site features should be displayed.

swansonk14 commented on web/config.py:
```diff
@@ -10,6 +10,7 @@
 
 _TEMP_FOLDER_OBJECT = TemporaryDirectory()
 
+DEMO = False
```
Would it be possible to move the demo flag to be a command line argument in `run.py` so that we don't have to hard code the website to be in demo mode?

lhirschfeld commented:
- Introduced 'DEMO' config variable, which decides whether all site features should be displayed.
- Added logic in templates to only show relevant information if in DEMO mode.
- Added wrapper for sensitive views, so that they will only perform standard functionality if not in DEMO mode.
